core/storage/exploration/gae_models.py

```diff
@@ -1113,3 +1135,121 @@ def get_export_policy(cls) -> Dict[str, base_models.EXPORT_POLICY]:
+        required=True, indexed=True)
+    # The details of the previous commit on each state at a particular
+    # version of the exploration. The json structure will look like the
+    # following:
+    # {
+    #   [state_name: str]: {
+    #     "previously_edited_in_version": int,
+    #     "state_name_in_previous_version": str,
+    #     "committer_id": str
+    #   }
+    # }
+    # The json object can have multiple keys in this case depending on
+    # the number of states.
+    state_version_history = datastore_services.JsonProperty(
+        default={}, indexed=False)
+    # The details of the previous commit on the exploration metadata at a
+    # particular version of the exploration. The json structure will look
+    # like the following:
+    # {
+    #   "previously_edited_in_version": str,
+    #   "committer_id": str.
+    # }
+    # The json object will have only two keys in this case.
+    metadata_version_history = datastore_services.JsonProperty(
+        default={}, indexed=False)
+    # The user ids of the users who did the 'previous commit' on each state
+    # in a particular version of the exploration. It is required during the
+    # wipeout process to query for the models efficiently.
+    committer_ids = datastore_services.StringProperty(
+        indexed=True, repeated=True)
+
+    @classmethod
+    def get_instance_id(cls, exp_id: str, exp_version: int) -> str:
+        """Returns ID of the exploration version history model.
+
+        Args:
+            exp_id: str. The ID of the exploration.
+            exp_version: int. The version of the exploration.
+
+        Returns:
+            str. A string containing exploration ID and
+            exploration version.
+        """
+        return 'version-history-%s-%s' % (exp_id, exp_version)
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id: str) -> bool:
+        """Check whether ExplorationVersionHistoryModel references
+        the given user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return ExplorationVersionHistoryModel.query(
+            ExplorationVersionHistoryModel.committer_ids == user_id
+        ).get(keys_only=True) is not None
+
+    @staticmethod
+    def get_deletion_policy() -> base_models.DELETION_POLICY:
+        """Model contains data to pseudonymize corresponding to a user:
+        committer_ids field and the user ids stored in state_version_history
+        or metadata_version_history.
+        """
+        return base_models.DELETION_POLICY.LOCALLY_PSEUDONYMIZE
+
+    @staticmethod
+    def get_model_association_to_user(
+    ) -> base_models.MODEL_ASSOCIATION_TO_USER:
+        """The version history of states or exploration metadata is not
+        relevant for the purposes of Takeout since they don't contain relevant
```

Comment:
I think it may be better to say that all this data (the versions a user committed to) is already contained in model XYZ that falls under Takeout? Ditto below.

Reply:
Done.